plugins/modules/ec2_imagebuilder_component.py

```diff
@@ -0,0 +1,399 @@
+            params, current_component_params):
+        connection.create_component(**params)
+        changed = True
+
+    # update tags if component exists and tags have changed
+    if current_component_params and have_component_tags_changed(params, current_component_params):
+        update_component_tags(connection, arn, params, current_component_params)
+        changed = True
+
+    # if a change has been made, refetch
+    if changed:
+        current_component_params = get_component(connection, module, arn)
+
+    module.exit_json(changed=changed, component=camel_dict_to_snake_dict(current_component_params))
+
+
+def delete_component(connection, module, arn, component):
+    """
+    Delete an EC2 Image Builder component if it exists
+
+    :param connection: AWS boto3 imagebuilder connection
+    :param module: Ansible module
+    :param component: a dict of EC2 Image Builder component representing current state of resource, or None
+    :return: True if component was deleted else False
+    """
+
+    changed = False
+
+    if component:
+        try:
+            connection.delete_component(
+                componentBuildVersionArn=arn)
+            changed = True
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json_aws(e)
+
+    module.exit_json(changed=changed, component=dict(arn=arn))
+
+
+def get_aws_account_id(module):
+    return module.client('sts').get_caller_identity()['Account']
+
+
+def get_component(connection, module, arn):
+    """
+    Get an EC2 Image Builder component by ARN. If not found, return None.
+
+    :param connection: AWS boto3 imagebuilder connection
+    :param module: Ansible module
+    :param arn: ARN of Image Builder component to get
+    :return: boto3 Image Builder component dict or None if not found
+    """
+
+    try:
+        return connection.get_component(componentBuildVersionArn=arn)['component']
+    except (BotoCoreError, ClientError) as e:
+        if e.response['Error']['Code'] == 'ResourceNotFoundException':
+            return None
+        else:
+            module.fail_json_aws(e)
+
+
+def has_component_params_changed(params, current_component_params):
+    """
+    Compare components (sans tags). If there is a difference, return True immediately else return False
+
+    :param params: the component params
+    :param current_component_params: the current state of the component params
+    :return: True if any parameter is mismatched else False
+    """
+
+    simple_comparison_keys = ['change_description', 'data',
+                              'description', 'kms_key_id', 'platform', 'semantic_version', 'uri']
+    for k in simple_comparison_keys:
+        if k in params and params[k] != current_component_params[k]:
+            return True
+
+    return False
+
+
+def have_component_tags_changed(params, current_component_params):
+    return 'tags' in params and params['tags'] != current_component_params['tags']
```

Comment:
I prefer to split this logic up a little for readability
```suggestion
    if 'tags' not in params:
        return False
    return params['tags'] != current_component_params['tags']
```

We also have a helper "compare_aws_tags" which would make it easier in future to support only updating the tags rather than always 'purging' them.

https://docs.ansible.com/ansible/latest/dev_guide/platforms/aws_guidelines.html#compare-aws-tags